packages/reframed/src/reframed.ts

```diff
@@ -16,16 +16,26 @@ export function reframed(
 	iframe: HTMLIFrameElement;
 	container: HTMLElement;
 	// TODO: revert back to Promise<void> once ready promise is cleaned up
-	ready: Promise<[() => void, void]>;
+	ready: Promise<() => void>;
 } {
+	const reframeMetadata: reframeMetadata = {
+		iframeDocumentReadyState: "loading",
+	};
+
 	// create the reframed container
-	const reframedContainer =
+	const reframedContainer = (
 		"container" in options
 			? options.container
-			: document.createElement(options.containerTagName);
-	const reframedContainerShadowRoot =
+			: document.createElement(options.containerTagName)
+	) as ReframedContainer;
+
+	const reframedContainerShadowRoot = Object.assign(
 		reframedContainer.shadowRoot ??
-		reframedContainer.attachShadow({ mode: "open" });
+			reframedContainer.attachShadow({ mode: "open" }),
+		{
+			reframeMetadata,
```

Comment:
Instead of monkey patching, could we use a symbol as a key here.

Please create the symbol via `Symbol("reframedMetadata");` rather than vai `Symbol.for`

Reply:
yeah! awesome point! I should be reaching out for symbols more often! 🙈😓 , I've made the change! thanks! 🙂 :+1: